### PR TITLE
Add tests to create a new project

### LINKED ISSUE #10: Add playwright tests for creation of a new project in neetoPlanner
### Steps:
1. Visit your neetoPlanner subdomain.
2. Click on the "Projects" section on the sidebar.
3. Click on the "Add new Project" button.
4. Fill in the Project name and description in the respective fields.
5. Click on the submit button.

### Expected Behavior:
- The Project should be successfully created.
- The user should be navigated to the Project details page after creation.

### Assertions:
- Verify that the Project is listed in the Projects section.
- Verify that clicking on the created Project navigates to its details page.

### PR DESCRIPTION
**Pull request by Srutip04**
Closes #10

### CODE REVIEW

File: e2e/poms/projects.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    addNewProjectAndVerify = async ({ projectName, projectDescription,projectIdentifier }: Project) => {
+        await this.page.getByRole('button', { name: BUTTON_LABELS.addNewProject }).click();
+        await this.page.getByPlaceholder(PLACEHOLDERS.projectName).fill(projectName);
+        await this.page.getByPlaceholder(PLACEHOLDERS.projectIdentifier).fill(projectIdentifier);
+        await this.page.getByPlaceholder(PLACEHOLDERS.projectDescription).fill(projectDescription);
+        await this.page.getByRole('button', { name: BUTTON_LABELS.saveChanges }).click();
+        await expect(this.page.getByText(new RegExp(projectName, "i"))).toBeVisible();
+        expect(this.page).toHaveURL(new RegExp(projectIdentifier.toUpperCase()));
```
Comment (shreyakurian02) [inline e2e/poms/projects.ts]:
```suggestion
       await  expect(this.page).toHaveURL(new RegExp(projectIdentifier.toUpperCase()));
```

File: e2e/tests/projects.spec.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    test.beforeEach(() => {
+        projectName = faker.word.words({ count: 5 });
+        projectDescription = faker.word.words({ count: 10 });
+        projectIdentifier = faker.string.alpha({ length: 5, casing: 'upper', exclude: ['A'] }) 
+    });
+
+    test("should create a new project", async ({ page, projectPage }) => {
+        await test.step("Step 1: Navigate to Base URL",  async () => {
+            await page.goto("/");
+        });
```
Comment (shreyakurian02) [inline e2e/tests/projects.spec.ts]: Avoid redundant async awaits. 
https://courses.bigbinaryacademy.com/learn-qa-automation-using-playwright/best-practices/#bigbinary-best-practices:~:text=5.-,Avoid,-redundant%20awaits%20when

**Review by shreyakurian02 — CHANGES_REQUESTED** [review]
@Srutip04 
It is important to destroy the resources that were created in a test right after they have been executed. Make the fix for the same along with the following suggestions.

**Comment by Srutip04** [comment]
I've addressed the requested changes in this PR.Please review the changes.

File: e2e/poms/projects.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    deleteProject = async ({projectName}) =>{
+        await expect(this.page.getByText(new RegExp(projectName, "i"))).toBeVisible();
+        await this.page.locator('button:nth-child(4)').click();
+        await this.page.getByRole('button', { name: BUTTON_LABELS.deleteButton }).click();
+        await expect(this.page.getByRole('heading', { name: HEADINGS.deleteProject })).toBeVisible();
+        await this.page.getByRole('button', { name: BUTTON_LABELS.deleteButton }).click();
+        await expect(this.page.getByRole('button', { name: BUTTON_LABELS.addNewProject })).toBeVisible();
+
+    }
+}
```
Comment (shreyakurian02) [inline e2e/poms/projects.ts]:
```suggestion
    addNewProjectAndVerify = async ({ projectName, projectDescription,projectIdentifier }: Project) => {
        await this.page.getByRole('button', { name: BUTTON_LABELS.addNewProject }).click();
        await this.page.getByPlaceholder(PLACEHOLDERS.projectName).fill(projectName);
        await this.page.getByPlaceholder(PLACEHOLDERS.projectIdentifier).fill(projectIdentifier);
        await this.page.getByPlaceholder(PLACEHOLDERS.projectDescription).fill(projectDescription);
        await this.page.getByRole('button', { name: BUTTON_LABELS.saveChanges }).click();
        await expect(this.page.getByText(new RegExp(projectName, "i"))).toBeVisible();
        await expect(this.page).toHaveURL(new RegExp(projectIdentifier.toUpperCase()));
    }
    
    deleteProject = async ({projectName}) => {
        await expect(this.page.getByText(new RegExp(projectName, "i"))).toBeVisible();
        await this.page.locator('button:nth-child(4)').click();
        await this.page.getByRole('button', { name: BUTTON_LABELS.deleteButton }).click();
        await expect(this.page.getByRole('heading', { name: HEADINGS.deleteProject })).toBeVisible();
        await this.page.getByRole('button', { name: BUTTON_LABELS.deleteButton }).click();
        await expect(this.page.getByRole('button', { name: BUTTON_LABELS.addNewProject })).toBeVisible();
    }
}
```

Comment (shreyakurian02) [reply e2e/poms/projects.ts]: Take care of unnecessary spacings.

File: e2e/poms/projects.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        await this.page.getByPlaceholder(PLACEHOLDERS.projectIdentifier).fill(projectIdentifier);
+        await this.page.getByPlaceholder(PLACEHOLDERS.projectDescription).fill(projectDescription);
+        await this.page.getByRole('button', { name: BUTTON_LABELS.saveChanges }).click();
+        await expect(this.page.getByText(new RegExp(projectName, "i"))).toBeVisible();
+        await expect(this.page).toHaveURL(new RegExp(projectIdentifier.toUpperCase()));
+
+    }
+    deleteProject = async ({projectName}) =>{
+        await expect(this.page.getByText(new RegExp(projectName, "i"))).toBeVisible();
+        await this.page.locator('button:nth-child(4)').click();
```
Comment (shreyakurian02) [inline e2e/poms/projects.ts:28]: https://courses.bigbinaryacademy.com/learn-qa-automation-using-playwright/best-practices/#bigbinary-best-practices:~:text=remain%20idle%20indefinitely.-,13,-.%20Avoid%20using%20nth

File: e2e/poms/projects.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        await this.page.getByRole('button', { name: BUTTON_LABELS.deleteButton }).click();
+        await expect(this.page.getByRole('heading', { name: HEADINGS.deleteProject })).toBeVisible();
+        await this.page.getByRole('button', { name: BUTTON_LABELS.deleteButton }).click();
```
Comment (shreyakurian02) [inline e2e/poms/projects.ts]: There is no need to assert for an element to be visible before clicking until absolute necessary.

File: e2e/poms/projects.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        await expect(this.page.getByRole('button', { name: BUTTON_LABELS.addNewProject })).toBeVisible();
```
Comment (shreyakurian02) [inline e2e/poms/projects.ts:31]: This is not the right way of checking if the project was deleted. What if in parallelism few other projects were created in other tests as well. This test would be highly flaky.

**Review by shreyakurian02 — APPROVED** [review]
@Srutip04 Make the changes and merge the PR.

**Review by shreyakurian02 — CHANGES_REQUESTED** [review]
@Srutip04

**Review by shreyakurian02 — CHANGES_REQUESTED** [review]
@Srutip04

**Review by shreyakurian02 — CHANGES_REQUESTED** [review]
@Srutip04

**Pull request merged**